core/src/main/scala/org/bykn/bosatsu/PackageMap.scala

```diff
@@ -52,7 +52,24 @@ object PackageMap {
   type MapF2[A, B] = MapF3[A, A, B]
   type ParsedImp = PackageMap[PackageName, Unit, Unit, (Statement, ImportMap[PackageName, Unit])]
   type Resolved = MapF2[Unit, (Statement, ImportMap[PackageName, Unit])]
-  type Inferred = PackageMap[Package.Interface, NonEmptyList[Referant[Variance]], Referant[Variance], Program[TypeEnv[Variance], TypedExpr[Declaration], Statement]]
+  type Typed[T] = PackageMap[
+    Package.Interface,
+    NonEmptyList[Referant[Variance]],
+    Referant[Variance],
+    Program[
+      TypeEnv[Variance],
+      TypedExpr[T],
+      Statement
+    ]
+  ]
+  
+  type Inferred = Typed[Declaration]
+
+  type Normalized = Typed[(Declaration, Normalization.NormalExpressionTag)]
```

Comment:
Maybe this should be a type defined in `object Normalization`. I like to avoid circular dependencies where possible, and I think putting this here makes PackageMap depend on Normalization, and also vice-versa, but maybe I'm wrong.

Reply:
I'm sure it does. I'll move it